### DISCUSSION TITLE
Difference in AR between iOS and Android

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#3552] [Q&A]
Hi guys, has anyone figured out a way to display models on iOS and Android so they look the same or at least close? I keep getting darker models (my guess is due to light estimation) on iOS and lighter models on Android (WebXR), its getting ridiculous at this point... just look at the difference... 

![ECA3D763-9DA3-4E76-A212-04D99E0EED56](https://user-images.githubusercontent.com/92090684/175954046-5281ed10-dff3-45af-bafb-1452ad5dae27.png)
![Screenshot_20220627-134508_Chrome](https://user-images.githubusercontent.com/92090684/175954055-8e62c646-ec4b-4ba7-a095-9bc54b7b83bb.jpg)

**Comment by elalish** [comment]
Agreed, this is super frustrating. First, Apple only uses USDZ and there isn't a clear way to convert glTF materials to USDZ, so that conversion tends to be best effort. Consistency will become a lot easier once Apple supports WebXR and/or glTF (@grorg). 

Can you list which AR mode is each picture and also which one you consider to be closer to the authored intent? Apple uses estimated lighting in AR, while WebXR does not by default (but you can try enabling it with `xr-environment`). You might also try Scene Viewer, which also uses ARCore's lighting estimation. With default WebXR mode, you can change the `exposure` or the `environment-image`. I'm curious if you find any of these to help.

**Reply by danilokrulj** [reply]
The top picture (darker model) is QuickLook and I would say its closer to the original sofa, and the bottom one is WebXR. I did try enabling xr-environment but its unpredictable and i feel like its light estimation is way off (I remember testing xr-environment and when I moved the model to a darker side of the room it got more light instead of less, also when i moved the model into the sunlight it became darker). I also played around with the exposure and the environment-image and it did help with the color difference but its still visible.

**Reply by elalish** [reply]
Thanks, that's helpful. Was the file originally authored in USDZ or glTF, or were they both exported from the same authoring tool?

**Reply by danilokrulj** [reply]
Its originally a gltf file (exported from blender) and in the picture i think the usdz was auto-generated, although i later converted the gltf into usdz manually (in realityconverter and xcode) to try and get a more similar result. I got closer by increasing the exposure before exporting the usdz, and decreasing the exposure in the viewer (so the gltf gets darker) but the difference is still visible. Can't wait for iOS to implement WebXR, will make my life a lot easier.

**Reply by elalish** [reply]
Interesting, it does beg the question of why the glTF (as originally authored) doesn't match your desired perception. I would focus on making that right first (via lighting and materials), as any secondary conversion to USDZ is going to introduce yet more errors and the fact you like it better this time is only by chance.
